services/wallet/router/router_v2.go

```diff
@@ -294,98 +322,141 @@ func findBestV2(routes [][]*PathV2, tokenPrice float64, nativeChainTokenPrice fl
-			return errors.New("username is required for ENSRelease")
+			return routererrors.ErrUsernameRequiredForENSRelease
 		}
 	}
 
 	if input.SendType == ENSSetPubKey {
 		if input.Username == "" || input.PublicKey == "" || ens.ValidateENSUsername(input.Username) != nil {
-			return errors.New("username and public key are required for ENSSetPubKey")
+			return routererrors.ErrUsernameAndPubKeyRequiredForENSSetPubKey
 		}
 	}
 
 	if input.SendType == StickersBuy {
 		if input.PackID == nil {
-			return errors.New("packID is required for StickersBuy")
+			return routererrors.ErrPackIDRequiredForStickersBuy
 		}
 	}
 
 	if input.SendType == Swap {
 		if input.ToTokenID == "" {
-			return errors.New("toTokenID is required for Swap")
+			return routererrors.ErrToTokenIDRequiredForSwap
 		}
 		if input.TokenID == input.ToTokenID {
-			return errors.New("tokenID and toTokenID must be different")
+			return routererrors.ErrTokenIDAndToTokenIDDifferent
 		}
 
 		// we can do this check, cause AmountIn is required in `RouteInputParams`
 		if input.AmountIn.ToInt().Cmp(pathprocessor.ZeroBigIntValue) > 0 &&
 			input.AmountOut != nil &&
 			input.AmountOut.ToInt().Cmp(pathprocessor.ZeroBigIntValue) > 0 {
-			return errors.New("only one of amountIn or amountOut can be set")
+			return routererrors.ErrOnlyOneOfAmountInOrOutSet
 		}
 
 		if input.AmountIn.ToInt().Sign() < 0 {
-			return errors.New("amountIn must be positive")
+			return routererrors.ErrAmountInMustBePositive
 		}
 
 		if input.AmountOut != nil && input.AmountOut.ToInt().Sign() < 0 {
-			return errors.New("amountOut must be positive")
+			return routererrors.ErrAmountOutMustBePositive
 		}
 	}
 
 	if input.FromLockedAmount != nil && len(input.FromLockedAmount) > 0 {
+		copyMap := func(src map[uint64]bool) map[uint64]bool {
+			dst := make(map[uint64]bool)
+			for key, value := range src {
+				dst[key] = value
+			}
+			return dst
+		}
+
+		suppNetworks := copyMap(supportedNetworks)
+		if input.TestnetMode {
+			suppNetworks = copyMap(supportedTestNetworks)
+		}
+
+		totalLockedAmount := big.NewInt(0)
+
 		for chainID, amount := range input.FromLockedAmount {
 			if input.TestnetMode {
 				if !supportedTestNetworks[chainID] {
-					return errors.New("locked amount is not supported for the selected network")
+					return routererrors.ErrLockedAmountNotSupportedForNetwork
 				}
 			} else {
 				if !supportedNetworks[chainID] {
-					return errors.New("locked amount is not supported for the selected network")
+					return routererrors.ErrLockedAmountNotSupportedForNetwork
 				}
 			}
 
+			delete(suppNetworks, chainID)
+
+			totalLockedAmount = new(big.Int).Add(totalLockedAmount, amount.ToInt())
+
 			if amount == nil || amount.ToInt().Sign() < 0 {
-				return errors.New("locked amount must be positive")
+				return routererrors.ErrLockedAmountMustBePositive
 			}
 		}
+
+		if totalLockedAmount.Cmp(input.AmountIn.ToInt()) > 0 {
+			return routererrors.ErrLockedAmountExceedsTotalSendAmount
```

Comment:
Can this be checked before? I mean in validation?

Reply:
It is already part of `validateInputData` function, which is the first one called.